internal/api/filter_tests.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/buildkite/test-splitter/internal/plan"
+)
+
+type FilterTestsParams struct {
+	Files          []plan.TestCase `json:"files"`
+	Parallelism    int             `json:"parallelism"`
+	SplitByExample bool            `json:"split_by_example"`
```

Comment:
```suggestion
	Parallelism    int             `json:"parallelism"`
	SplitByExample bool            `json:"split_by_example"`
```
The server won't do anything with this information yet, but these are the least information we need to support dynamic slow file threshold, and quarantine.

Reply:
I feel like we had landed on sending the sanitized list of environment variables as part of this payload instead of defining a custom struct.

Reply:
What's the reasoning behind sending the environment variables? Is that so we don't need to update the client if we need more information, or to avoid translating env and struct?

Reply:
It's the former yeah.